integrated_channels/degreed/client.py

```diff
@@ -0,0 +1,213 @@
+# -*- coding: utf-8 -*-
+"""
+Client for connecting to Degreed.
+"""
+
+from __future__ import absolute_import, unicode_literals
+
+import datetime
+import time
+
+import requests
+
+from django.apps import apps
+
+from six.moves.urllib.parse import urljoin  # pylint: disable=import-error
+
+
+class DegreedAPIClient(object):
+    """
+    Client for connecting to Degreed.
+
+    Specifically, this class supports obtaining access tokens and posting to the courses and
+    completion status endpoints.
+    """
+
+    CONTENT_PROVIDER_SCOPE = 'provider_content'
+    COMPLETION_PROVIDER_SCOPE = 'provider_completion'
+    SESSION_TIMEOUT = 60
+
+    def __init__(self, enterprise_configuration):
+        """
+        Instantiate a new client.
+
+        Args:
+            enterprise_configuration (DegreedEnterpriseCustomerConfiguration): An enterprise customers's
+            configuration model for connecting with Degreed
+
+        Raises:
+            ValueError: If a URL or access token are not provided.
+        """
+
+        if not enterprise_configuration:
+            raise ValueError('An DegreedEnterpriseCustomerConfiguration must be supplied!')
+
+        self.global_degreed_config = apps.get_model('degreed', 'DegreedGlobalConfiguration').current()
+        self.enterprise_configuration = enterprise_configuration
+        self.session = None
+        self.expires_at = None
+
+    def send_completion_status(self, payload):
+        """
+        Send a completion status payload to the Degreed Completion Status endpoint
+
+        Args:
+            payload: JSON encoded object (serialized from DegreedLearnerDataTransmissionAudit)
+                containing completion status fields per Degreed documentation.
+
+        Returns:
+            A tuple containing the status code and the body of the response.
+        Raises:
+            HTTPError: if we received a failure response code from Degreed
+        """
+        return self._post(
+            urljoin(self.global_degreed_config.degreed_base_url, self.global_degreed_config.completion_status_api_path),
+            payload,
+            self.COMPLETION_PROVIDER_SCOPE
+        )
+
+    def delete_completion_status(self, payload):
```

Comment:
Same here about the name `delete_course_completion`.

Reply:
:+1: